presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/thrift/DefaultThriftMetastoreClientFactory.java

```diff
@@ -120,16 +126,28 @@ private TTransport createTransport(HostAndPort address, Optional<String> delegat
         try {
             // load KeyStore if configured and get KeyManagers
             KeyManager[] keyManagers = null;
+            KeyStore keyStore = null;
+            char[] keyManagerPassword = new char[0];
             if (keyStorePath.isPresent()) {
-                KeyStore keyStore = PemReader.loadKeyStore(keyStorePath.get(), keyStorePath.get(), keyStorePassword);
+                try {
+                    keyStore = PemReader.loadKeyStore(keyStorePath.get(), keyStorePath.get(), keyStorePassword);
+                }
+                catch (IOException | GeneralSecurityException e) {
+                    log.warn(e, "Failed reading keystore as PEM file");
```

Comment:
I don't think we need a warning as this is expected if the file is a key store. I would just drop the message entirely.